docs/book/src/clusterctl/provider-contract.md

```diff
@@ -241,6 +248,47 @@ Templates writers should use the common variables to ensure consistency across p
 Additionally, the value of the command argument to `clusterctl generate cluster <cluster-name>` (`<cluster-name>` in this case), will
 be applied to every occurrence of the `${ CLUSTER_NAME }` variable.
 
+### ClusterClass definitions
+
+An infrastructure provider could publish a **ClusterClass definition** file to be used by `clusterctl generate cluster` that will be used along
+with the workload cluster templates.
+This is a single YAML with _all_ the objects required that make up the ClusterClass.
+
+The following rules apply:
+
+#### Naming conventions
+
+ClusterClass definitions MUST be stored in the same folder as the component YAML and follow this naming convention:
+1. The ClusterClass definition should be named `clusterclass-{ClusterClass-name}.yaml`, e.g `clusterclass-prod.yaml`.
+
+`{ClusterClass-name}` is the name of the ClusterClass that is referenced from the Cluster.spec.topology.class field 
+in the Cluster template; Cluster template files using a ClusterClass are usually simpler because they are no longer 
+required to have all the templates.
+
+Each provider should create user facing documentation with the list of available ClusterClass definitions.
+
+#### Target namespace
+
+The ClusterClass definition YAML MUST assume the target namespace already exists.
+
+The references in the ClusterClass definition should NOT specify a namespace.
+
+It is recommended that none of the objects in the ClusterClass YAML should specify a namespace.
+
+Even if technically possible, it is strongly recommended that none of the objects in the ClusterClass definitions are shared acorss multiple definitions; 
+this helps in preventing changing an object inadvertently impacting many ClusterClasses, and consequently, all the Clusters using those ClusterClasses.
+
+#### Variables
+
+Currently the ClusterClass definitions SHOULD NOT have any variables in them.
+
+ClusterClass definitions files should not use variable substitution, given that ClusterClass and managed topologies provide an alternative model for variable definion. 
```

Comment:
```suggestion
Currently the ClusterClass definitions SHOULD NOT have any variables in them.

ClusterClass definitions files should not use variable substitution, given that ClusterClass and managed topologies provide an alternative model for variable definition. 
```

Reply:
I read it as the first sentence mentioning that ClusterClasses should not have variables (envsubst style variables as used in workload cluster templates) in them and the second sentence explains why we should not use variables in ClusterClass definitions. 

>Not sure if it is intended, but this sounds like the ClusterClasses should also not use variables (the new ones) as it writes that variables and variable substitutions should not be used.

By "the new ones" I am assuming you mean the variables used by for patching. If so, the doc is not talking about them.
How about "Currently the ClusterClass definitions SHOULD NOT have any environment variables in them." to avoid ambiguity.?

Reply:
Sounds great! My problem was only that the there's no way to tell with the current wording if the variables in the first sentence are ClusterClass variables.